pyro/distributions/hmm.py

```diff
@@ -980,7 +983,8 @@ def log_prob(self, value):
 
     def rsample(self, sample_shape=torch.Size()):
         assert self.duration is not None
-        init = self.initial_dist.rsample(sample_shape)
+        # XXX we might not need to expand here because we already expand eagerly in the construction
+        init = self.initial_dist.expand(self.batch_shape).rsample(sample_shape)
```

Comment:
should we remove `expand` here and the following ones?

Reply:
Sure, but could you add assertions in case the `__init__()` method later chages?
```py
assert self.initial_dist.batch_shape == self.batch_shape
assert self.transition_dist.batch_shape == self.batch_shape + (self.duration,)
assert self.observation_dist.batch_shape == self.batch_shape + (self.duration,)
```

Reply:
Sorry for the confusion, I have reverted the changes. `expand` is required in the case: those trans_dist, obs_dist has time_dim=1, but duration > 1.